pkg/hub/submarinerbroker/controller.go

```diff
@@ -174,6 +174,10 @@ func (c *submarinerBrokerController) reconcileManagedClusterSet(ctx context.Cont
 		return c.doClusterSetCleanup(ctx, clusterSet, recorder)
 	}
 
+	if isClusterSetEmpty(clusterSet) && finalizer.IsPresent(clusterSet, brokerFinalizer) {
+		return c.deleteBrokerResources(ctx, clusterSet, recorder)
+	}
```

Comment:
When an `ManagedClusterSet` (MCS) is first created it won't have any `ManagedClusters` (MC). So after we add the finalizer below, it returns and this will prevent subsequent updates from applying the broker resources until the MCS is no longer empty. This seems fine.

It would be useful to add unit tests for this flow. Since I'm familiar with the tests I can add them if you want.

Reply:
currently finalizer is added on clustersets, even when they are empty and submariner is not installed. e.g. `default` clusterset. A better check would allow us to add a `Info` log here without it getting spammed by other empty clustersets.